tests/test_satimg.py

```diff
@@ -23,6 +22,34 @@ def test_load_subclass(self):
         img = si.SatelliteImage(fn_img, read_from_fn=False)
         img = si.SatelliteImage(fn_img)
 
+    def test_copy(self):
+
+        fn_img = datasets.get_path("landsat_B4")
+
+        img = si.SatelliteImage(fn_img)
+
+        img2 = copy.copy(img)
+        img3 = img.copy()
+
+        # check the copied object is indeed a SatelliteImage
+        assert isinstance(img2,geoutils.satimg.SatelliteImage)
+        assert isinstance(img3,geoutils.satimg.SatelliteImage)
+
+        # check all immutable attributes are equal
+        georaster_attrs = ['bounds', 'count', 'crs', 'dtypes', 'height', 'indexes', 'nodata',
+                         'res', 'shape', 'transform', 'width']
+        satimg_attrs =  ['satellite', 'sensor', 'product', 'version', 'tile_name', 'datetime']
+
+        all_attrs = georaster_attrs + satimg_attrs
+
+        for attr in all_attrs:
+            assert img2.__getattribute__(attr) == img.__getattribute__(attr)
+
+        # check data array
+        assert np.count_nonzero(~img.data == img2.data) == 0
```

Comment:
again, you can use np.all(img.data == img2.data) which is easier to understand

Reply:
May I suggest `np.array_equal(img.data, img2.data, equal_nan=True)` instead?
This is why:
```python
In [1]: arr1 = np.zeros(3)

In [2]: arr1[0] = np.nan

In [3]: arr2 = arr1.copy()

In [4]: arr1 == arr2
Out[4]: array([False,  True,  True])

In [5]: np.all(arr1 == arr2)
Out[5]: False
```
The `__eq__` method in Numpy arrays doesn't like NaNs!

Reply:
Done everywhere.